oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/CompressedDocumentPropertyState.java

```diff
@@ -0,0 +1,350 @@
+        }
+    }
+
+    /**
+     * Read a multi valued {@code PropertyState} from a {@link JsopReader}.
+     *
+     * @param name the name of the property state
+     * @param reader the reader
+     * @return new property state
+     */
+    PropertyState readArrayProperty(String name, JsopReader reader) {
+        return readArrayProperty(name, store, reader);
+    }
+
+    /**
+     * Read a multi valued {@code PropertyState} from a {@link JsopReader}.
+     *
+     * @param name the name of the property state
+     * @param store the store
+     * @param reader the reader
+     * @return new property state
+     */
+    static PropertyState readArrayProperty(String name, DocumentNodeStore store, JsopReader reader) {
+        int type = PropertyType.STRING;
+        List<Object> values = Lists.newArrayList();
+        while (!reader.matches(']')) {
+            if (reader.matches(JsopReader.NUMBER)) {
+                String number = reader.getToken();
+                Long maybeLong = LongUtils.tryParse(number);
+                if (maybeLong == null) {
+                    type = PropertyType.DOUBLE;
+                    values.add(Double.parseDouble(number));
+                } else {
+                    type = PropertyType.LONG;
+                    values.add(maybeLong);
+                }
+            } else if (reader.matches(JsopReader.TRUE)) {
+                type = PropertyType.BOOLEAN;
+                values.add(true);
+            } else if (reader.matches(JsopReader.FALSE)) {
+                type = PropertyType.BOOLEAN;
+                values.add(false);
+            } else if (reader.matches(JsopReader.STRING)) {
+                String jsonString = reader.getToken();
+                int split = TypeCodes.split(jsonString);
+                if (split != -1) {
+                    type = TypeCodes.decodeType(split, jsonString);
+                    String value = TypeCodes.decodeName(split, jsonString);
+                    if (type == PropertyType.BINARY) {
+                        values.add(store.getBlobFromBlobId(value));
+                    } else if (type == PropertyType.DOUBLE) {
+                        values.add(Conversions.convert(value).toDouble());
+                    } else if (type == PropertyType.DECIMAL) {
+                        values.add(Conversions.convert(value).toDecimal());
+                    } else {
+                        values.add(StringCache.get(value));
+                    }
+                } else {
+                    type = PropertyType.STRING;
+                    values.add(StringCache.get(jsonString));
+                }
+            } else {
+                throw new IllegalArgumentException("Unexpected token: " + reader.getToken());
+            }
+            reader.matches(',');
+        }
+        return createProperty(name, values, Type.fromTag(type, true));
+    }
+}
+
+
+
+
+
+
+
```

Comment:
detail but we probably should remove those lines

Reply:
Done